third-party/realdds/src/dds-option.cpp

```diff
@@ -46,15 +145,345 @@ void dds_option::init_stream( std::shared_ptr< dds_stream_base > const & stream
+    }
+
+    // Finally, set the actual value
+    option->init_value( value );
+    // (and the option is now initialized)
+
+    return option;
+}
+
+
+json dds_option::to_json() const
+{
+    json j = json::array();
+    j += get_name();
+    j += get_value();
+    if( ! _minimum_value.is_null() || ! _maximum_value.is_null() || ! _stepping.is_null() )
+    {
+        j += _minimum_value.is_null() ? rsutils::null_json : _minimum_value;
+        j += _maximum_value.is_null() ? rsutils::null_json : _maximum_value;
+        j += _stepping.is_null() ? rsutils::null_json : _stepping;
+    }
+    if( get_default_value().exists() )
+        j += get_default_value();
+    j += get_description();
+
+    auto props = props_to_json();
+    if( ! props.empty() )
+        j += props;
+
+    return j;
+}
+
+
+json dds_option::props_to_json() const
+{
+    json props = json::array();
+    if( is_optional() )
+        props += "optional";
+    if( is_read_only() && get_default_value().exists() )
+        props += "read-only";
+    return props;
+}
+
+
+/*static*/ dds_float_option::type dds_float_option::check_float( json const & value )
+{
+    try
+    {
+        // 1.1 is expressed as 1.10000002 as float but 1.1000000000000001 in double...
+        // Really hard to check that we don't lose precision unless we convert back to string which will be rounded
+        // anyway.
+        return value.get< type >();
+    }
+    catch( ... )
+    {
+    }
+    DDS_THROW( runtime_error, "not convertible to a float: " << value );
+}
+
+
+void dds_float_option::check_type( json const & value ) const
+{
+    check_float( value );
+}
+
+
+/*static*/ dds_integer_option::type dds_integer_option::check_integer( json const & value )
+{
+    switch( value.type() )
+    {
+    case json::value_t::number_integer:
+        return value.get< type >();
+
+    case json::value_t::number_unsigned:
+        if( value.get< json::number_unsigned_t >() > (uint64_t)std::numeric_limits< type >::max() )
+            break;
+        return value.get< type >();
+
+    case json::value_t::number_float:
+        if( value.get< type >() != value.get< json::number_float_t >() )
+            break;
+        return value.get< type >();
+    }
+    DDS_THROW( runtime_error, "not convertible to a signed integer: " << value );
+}
+
+
+void dds_integer_option::check_type( json const & value ) const
+{
+    check_integer( value );
+}
+
+
+void dds_string_option::check_type( json const & value ) const
+{
+    if( ! value.is_string() )
+        DDS_THROW( runtime_error, "not a string: " << value );
+}
+
+
+/*static*/ rsutils::string::ip_address dds_ip_option::check_ip( json const & value )
+{
+    ip_address ip( value.string_ref() );
+    if( ip.is_valid() )
+        return ip;
+    DDS_THROW( runtime_error, "not an IP address: " << value );
+}
+
+
+void dds_ip_option::check_type( json const & value ) const
+{
+    if( ! ip_address( value.string_ref() ).is_valid() )
```

Comment:
Why not calling `check_ip` as the other types? Looks like code duplication

Reply:
Done